packages/cardano-graphql/src/StakePoolSearchProvider/StakePoolSearchClient.ts

```diff
@@ -0,0 +1,96 @@
+import { ProviderError, ProviderFailure, StakePool } from '@cardano-sdk/core';
+import { gql, GraphQLClient } from 'graphql-request';
+import { StakePoolsQueryResponse } from './types';
+
+export class StakePoolSearchClient {
```

Comment:
This could just be a stateless function, separating the type and calling it `createStakePoolSearchClient`